packages/integration-sdk-runtime/src/synchronization/__tests__/shrinkBatchRawData.test.ts

```diff
@@ -261,4 +261,74 @@ describe('shrinkBatchRawData', () => {
     expect(logger.info).toBeCalledTimes(1);
     expect(logger.info).toHaveBeenCalledWith('Attempting to shrink rawData');
   });
+
+  it('should assign 0 to the size of an undefined property when building the property map', () => {
+    const largeData = new Array(700000).join('aaaaaaaaaa');
+    const data = [
+      {
+        _class: 'test',
+        _key: 'testKey',
+        _type: 'testType',
+        _rawData: [
+          {
+            name: 'test',
+            rawData: {
+              largeRawDataProp: largeData + 'more',
+              testRawData: 'test123',
+              testFinalData: 'test789',
+              anotherLargeRawDataProp: largeData,
+            },
+          },
+        ],
+      },
+      {
+        _class: 'test',
+        _key: 'testKey2',
+        _type: 'testType',
+        largeProperty: largeData,
+        undefProp: undefined,
+        falseyProp: false,
+        _rawData: [
+          {
+            name: 'test2',
+            rawData: {
+              testRawData: 'test123',
+              testFinalData: 'test789',
+            },
+          },
+        ],
+      },
+    ];
+    try {
+      shrinkBatchRawData(data, logger);
+    } catch (err) {
+      expect(err).toBeInstanceOf(IntegrationError);
+      expect(logger.error).toBeCalledTimes(1);
```

Comment:
This is a separate issue, but I kind of wish that this function didn't have access to the logger at all. It would be much cleaner if relevant properties were attached to the thrown error object. The logging could happen by the caller instead of inside of this utility. Separation of concerns!